tags/TagSet.md

```diff
@@ -0,0 +1,109 @@
+# Summary
+A `Tag` is used to label anything that is associated
+with a specific operation, such as an HTTP request. These `Tag`s are used to
+aggregate measurements in a [`View`](https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/DataAggregation.md#view) 
+according to unique value of the `Tag`s. The `Tag`s can also be used to filter (include/exclude)
+measurements in a `View`. `Tag`s can further be used for logging and tracing.
+
+# Tag
+A `Tag` consists of TagScope, TagKey, and TagValue.
+
+## TagKey
+
+`TagKey` is the name of the Tag. TagKey along with `TagValue` is used to aggregate
+and group stats, annotate traces and logs.
+
+**Restrictions**
+- Must contain only printable ASCII (codes between 32 and 126 inclusive with below Exception)
+- - Exceptions '=', ','
+- Must have length greater than zero and less than 256 combined with the source and metadata.
+- Must not be empty.
+
+## TagValue
+
+`TagValue` is a string or string wrapper. It MUST contain only printable ASCII (codes between
+32 and 126)
+
+## TagScope
+
+`TagScope` is used to determine the scope of a `Tag`. For now, it has two possible values,
+Local or Non-Local. In future, additional values be added to address specific situations.
+
+The tag creator determines the scope of the tag.
+
+**Local Scope**
+Tag with `Local` scope are used within the process it created. Such tags are not propagated
+across process boundaries. Even if the process is reentrant the tag MUST be excluded from
+propagation when the call leaves the process.
+
+**Non-Local Scope**
+If a tag is created with the `Non-Local` scope then it is propagated across process boundaries subject 
+to outgoing and incoming (on remote side) filter criteria. See `TagFilter` in [Tag Propagation](#Tag Propagation).
+
+# TagSet 
```

Comment:
Should this be TagMap as we discussed? Same for the file name.

Reply:
earlier one of the comment was to use TagSet. I can change it back to TagMap.